tests/unit/package_managers/test_pip.py

```diff
@@ -3415,7 +3414,24 @@ def test_download_dependencies(
                 pip.DistributionPackageInfo("foo", "1.0", "wheel", w3_path, "", False),
             ]
 
-        mock_distributions.return_value = source_package, wheels
+        pypi_downloads = [
+            sdist.download_info
+            | {
+                "kind": "pypi",
+                "hash_verified": use_hashes,
+                "requirement_file": str(req_file.file_path.subpath_from_root),
+            }
+        ] + [
+            wheel.download_info
+            | {
+                "kind": "pypi",
+                "hash_verified": use_hashes,
+                "requirement_file": str(req_file.file_path.subpath_from_root),
+            }
+            for wheel in wheels
+        ]
+
```

Comment:
You can't generalize all wheels like this, especially not the `hash_verified` part because each wheel is supposed to test a different behaviour (matching hash, mismatching hash, no hash), so this is another issue causing `test_download_dependencies` to fail.
Moreover, there's too much happening in this statement, please unwind/open code the logic ending with a simple `pypi_downloads = [sdist] + wheels`